frontend/src/views/ddahs/import-export.tsx

```diff
@@ -26,6 +27,35 @@ import { ddahTableSelector } from "../ddah-table/actions";
 import { ActionButton } from "../../components/action-buttons";
 import { FaDownload } from "react-icons/fa";
 
+/**
+ * A factory function which produces ddah prepareData function,
+ *
+ * @export
+ * @param {Function | null} ddahFilter
+ * @returns {Function}
+ */
+export function prepareDataFactory(ddahFilter?: Function) {
+    // Make a function that converts a list of ddahs into a `File` object.
+    return function prepareData(
+        ddahs: Ddah[],
+        dataFormat: "csv" | "json" | "xlsx"
+    ) {
+        if (ddahFilter && ddahFilter instanceof Function) {
```

Comment:
If `ddahFilter == false` it will not be an instance of `Function`, so you don't need the first check.